src/main/java/it/pagopa/pm/gateway/config/ClientConfig.java

```diff
@@ -65,54 +94,59 @@ public Jaxb2Marshaller jaxb2Marshaller() {
         return new AzureLoginClient();
     }
 
     @Bean
     public PaymentManagerControllerApi postePayControllerApi() {
-  log.info("START postePayControllerApi()");
+        log.info("START postePayControllerApi()");
         ApiClient apiClient = addProxyToApiClient(new ApiClient()
                 .setBasePath(POSTEPAY_CLIENT_URL)
                 .setConnectTimeout(POSTEPAY_CLIENT_TIMEOUT));
-       log.info("END - postePayControllerApi()- POSTEPAY_CLIENT_URL: " + POSTEPAY_CLIENT_URL);
+        log.info("END - postePayControllerApi()- POSTEPAY_CLIENT_URL: " + POSTEPAY_CLIENT_URL);
         return new PaymentManagerControllerApi(apiClient);
     }
 
     @Bean
-    public WebServiceTemplate bancomatPayWebServiceTemplate() {
+    public WebServiceTemplate bancomatPayWebServiceTemplate() throws RestApiException {
+        Map<String, String> bancomatClientConfigValues = getBancomatClientConfigValues();
+        String url = bancomatClientConfigValues.get(BPAY_URL);
+
         WebServiceTemplate webServiceTemplate = new WebServiceTemplate();
         webServiceTemplate.setMarshaller(jaxb2Marshaller());
         webServiceTemplate.setUnmarshaller(jaxb2Marshaller());
-        webServiceTemplate.setDefaultUri(BPAY_CLIENT_URL);
+        webServiceTemplate.setDefaultUri(url);
         for (WebServiceMessageSender sender : webServiceTemplate.getMessageSenders()) {
-            Duration durationTimeout = Duration.ofMillis(BPAY_CLIENT_TIMEOUT_MS);
+            Duration durationTimeout = Duration.ofMillis(Long.parseLong(bancomatClientConfigValues.get(TIMEOUT_MS)));
             if (sender instanceof HttpUrlConnectionMessageSender) {
                 ((HttpUrlConnectionMessageSender) sender).setConnectionTimeout(durationTimeout);
                 ((HttpUrlConnectionMessageSender) sender).setReadTimeout(durationTimeout);
             }
         }
-        log.info("bancomatPayWebServiceTemplate - bancomatPayClientUrl " + BPAY_CLIENT_URL);
+        log.info("bancomatPayWebServiceTemplate - bancomatPayClientUrl " + url);
         return webServiceTemplate;
     }
 
     @Bean
-    public RestTemplate microsoftAzureRestTemplate() {
+    public RestTemplate microsoftAzureRestTemplate() throws RestApiException {
+        Map<String, String> azureAuthClientConfigValues = getAzureAuthClientConfigValues();
+
         HttpComponentsClientHttpRequestFactory httpComponentsClientHttpRequestFactory =
                 new HttpComponentsClientHttpRequestFactory();
         httpComponentsClientHttpRequestFactory.setHttpClient(
                 HttpClientBuilder.create()
                         .setProxy(createProxy(this.getClass().getName()))
                         .setConnectionManager(createConnectionManager(
-                                AZURE_CLIENT_MAX_TOTAL,
-                                AZURE_CLIENT_MAX_PER_ROUTE))
+                                Integer.parseInt(azureAuthClientConfigValues.get(MAX_TOTAL)),
```

Comment:
Maybe you can create a custom string-integer map to be used without always parsing the string value

Reply:
It's feasable but values still have to be parsed before being put in the Map<String, Integer> since they come in String format

Reply:
Yes, that's what I meant before.
Anyway, it is not that useful here since these properties are read only once...